config/dynamic.c

```diff
@@ -138,7 +138,7 @@ process_client_message(int conn_fd,
 	ret = luaL_loadbuffer(lua_state, msg, msg_len, "message")
 		|| lua_pcall(lua_state, 0, 1, 0);
 	if (ret != 0) {
-		reply_msg = luaL_checklstring(lua_state, -1, &reply_len);
+		reply_msg = lua_tolstring(lua_state, -1, &reply_len);
```

Comment:
Notice that `lua_tolstring()` returns `NULL` when an error occurs. Is the code ready for this case? Does `lua_tolstring()` also set `reply_len` to zero in this case? A solution may be to test if the return is `NULL`, if so set `reply_msg` to `""`, and `reply_len` to zero.

Reply:
In fact, if we do set `reply_msg` to some string, it must be to an error message since we are handling an error condition here.

Reply:
I see as fine not login because error messages should be smallish and not being truncated anyway. Given that both cases return a string, the `if` statement is redundant. Do you see any case that would not be handled fine after removing the `if` statement?

Reply:
No, if we make the assumption that error strings will not need to be truncated then I think all cases will be handled.